### ISSUE TITLE
Files getting erroneously deduped when using fullPaths.

### ISSUE THREAD

**Issue by jkyle** [substack/node-browserify#940]
https://github.com/substack/node-browserify/pull/917 introduced a bug where certain files get erroneously deduped. For example if you have an `index.js` that's just `module.exports = require('./lib')` in multiple libraries, they all get deduped to the first occurrence of the file. So, you end up requiring the wrong `lib` in the other libraries.

```
/test_library_one
    /lib
    index.js // module.exports = require('./lib')
/test_library_two
    /lib
    index.js // module.exports = require('./lib')
```

In this case, the bundle entry for `/test_library_two/index.js` ends up being

``` javascript
"/path/to/test_library_two/index.js":[function(require,module,exports){
    module.exports=require("/path/to/test_library_one/index.js")
}
```

which ends up requiring `test_library_one`'s `/lib` into `test_library_two`.

I first noticed this with https://github.com/hapijs/joi. Browserify 5.12.0 works, but 5.12.1 breaks. It also works fine without using fullPaths, but watchify requires that option.

**Comment by saschagehlich** [comment]
@substack Still failing for me - see my last comment in this issue. :(

**Comment by Delagen** [comment]
Still not fixed due #1007
